src/main/java/MongoConnection.java

```diff
@@ -3,27 +3,51 @@
 import com.mongodb.client.MongoClients;
 import com.mongodb.client.MongoDatabase;
 
+/**
+ * jdoc description
+ * */
 public class MongoConnection
 {
-    private static MongoConnection mongoConnection = null;
+    private static boolean TestEnvironment = true;
+    private static MongoConnection mongoTestConnection = null;
```

Comment:
Code will be cleaner if there is only one mongoConnection (Which is a testConnection when environment says so)